### ISSUE TITLE
Develop a UI/GUI for file downloading

### ISSUE THREAD

**Issue by wacedungeoner** [wacedungeoner/wace#1]
Currently the only way to download files is to download the entire (potentially outdated) ZIP file per-subject from the Releases section, or to use an external website like the one linked in the README. The repository would be easier to use if there was a more simple directory tree than GitHub's, and if individual files/folders could be more easily downloaded.

This could also be integrated with the bot, which is currently unused.

**Comment by denosawr** [comment]
Thinking that the best course of action would be using GH Pages. Could set up an Action to automatically, on commit, generate a static file index and either push to a static file host or a dedicated `gh-pages` branch.

Shouldn't take too long, hoping to push something if I have time over the mid-semester break?

**Comment by denosawr** [comment]
> Thinking that the best course of action would be using GH Pages. Could set up an Action to automatically, on commit, generate a static file index and either push to a static file host or a dedicated `gh-pages` branch.
> 
> Shouldn't take too long, hoping to push something if I have time over the mid-semester break?

Bulk download could be handled by just linking to a tool like DownGit[1] or such. Alternatively, zip download could be handled locally by the front-end, as it does on the WACE archive.

[1]: https://downgit.github.io/

**Comment by denosawr** [comment]
Is there a design which you think would be good?

Examples: https://vce.rocks/files/exams/, https://hscpastpapers.com/

**Comment by wacedungeoner** [comment]
> Is there a design which you think would be good?
> 
> Examples: https://vce.rocks/files/exams/, https://hscpastpapers.com/

The HSC past papers one looks great, if you have the time to do that much.
Your own archive is also pretty good.

**Comment by denosawr** [comment]
Hmm in order to do such a layout, we would effectively require a directory tree to be generated which can be dynamically loaded and displayed by a front-end application - not sure if I have the time to work on that actually! For now I might just make a barebones UI (each folder is its own page etc) and others can contribute if they're interested.

**Comment by wacedungeoner** [comment]
> Hmm in order to do such a layout, we would effectively require a directory tree to be generated which can be dynamically loaded and displayed by a front-end application - not sure if I have the time to work on that actually! For now I might just make a barebones UI (each folder is its own page etc) and others can contribute if they're interested.

That's fine, a static UI should work fine for now.
If we do end up needing another folder folder (eg. geography/y12/sem2 exams), would we just have to manually add that to the UI? If so that shouldn't be too much effort to fix if we do end up making a new folder.

**Comment by skybird23333** [comment]
> Hmm in order to do such a layout, we would effectively require a directory tree to be generated which can be dynamically loaded and displayed by a front-end application - not sure if I have the time to work on that actually! For now I might just make a barebones UI (each folder is its own page etc) and others can contribute if they're interested.

I could probably help with this, we could just have a JSON file as a map for all the files and perhaps have pull requests processed(i.e uploads and changes to the map) via GH Actions.

**Comment by wace-vault** [comment]
> I could probably help with this, we could just have a JSON file as a map for all the files and perhaps have pull requests processed(i.e uploads and changes to the map) via GH Actions.

We already have it up and running on the gh-pages branch at [wace-vault.github.io/collection/](https://wace-vault.github.io/collection/), although if you want you can try to improve upon it. Currently there's not much CSS; looks a bit basic. Keeping the issue open for now for this reason.

**Comment by skybird23333** [comment]
Yes, I've seen it. I'm currently working on a new one based on a Vue app
over at my fork.

wace-vault ***@***.***> 于 2023年5月18日周四 17:54写道：

**Comment by wace-vault** [comment]
Partly completed by #13, bulk-downloading is still needed however.
